src/main/java/fact/io/hdureader/Fits.java

```diff
@@ -0,0 +1,194 @@
+package fact.io.hdureader;
+
+import com.google.common.io.ByteStreams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+
+/**
+ *
+ * A Fits object containing all HDUs in a given file.
+ * HDUs can be accessed by their name.
+ *
+ *    Fits f = Fits.fromPath(p)
+ *    HDU events = f.getHDU("ZDrsCellOffsets");
+ *    BinTable binTable = events.getBinTable();
+ *
+ * BinTables can also directly be accessed by name if they exist.
+ *
+ *    Fits.fromPath(p).getBinTableByName("Events")
+ *                    .ifPresent(binTable -> {
+ *                        //do something with bintable
+ *                    })
+ *
+ * Data from the BinTable can be read using the BinTableReader and ZFitsHeapReader classes.
+ *
+ *    Fits f = new Fits(u);
+ *    BinTable events = f.getBinTableByName("Events").orElseThrow(IOException::new);
+ *
+ *    for(OptionalTypesMap<String, Serializable> p : BinTableReader.forBinTable(events)){
+ *      assertTrue(p.containsKey("Data"));
+ *    }
+ *
+ *
+ * Created by mackaiver on 03/11/16.
+ */
+public class Fits {
+    private static Logger log = LoggerFactory.getLogger(Fits.class);
+
+
+    private final URL url;
+    private final List<HDU>  hdus = new ArrayList<>();
+
+    private Map<String, HDU> hduNames = new HashMap<>();
+
+    public HDU primaryHDU;
+
+    /**
+     * Creates a fits instance from a path object without throwing checked exceptions immediately.
```

Comment:
I don't understand this `without throwing checked exceptions immediately.` part what are *checked* exceptions?